frontend/src/resetUsername.js

```diff
@@ -1,37 +1,117 @@
-import {Logo, Textbox, FormButton} from "./resetPswd.js";
+import {useState, useEffect} from 'react';
 import FormCSS from "./form.module.css";
+import axios from 'axios';
+
+
 
 export default function ResetUsernameForm() {
+    const [email, setEmail] = useState('');
+    const [username, setUsername] = useState('');
+    const [newUsername, setNewUsername] = useState('');
+    const [message, setMessage] = useState('');
+
+    const checkUsername = async () => {
+        try {
+            const response = await axios.get(`http://localhost:8080/users/${newUsername}`)
+            return response.data.exists;
+        } catch (err) {
+            console.error('error checking username: ', err)
+            return false;
+        }
+    }
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        setMessage('');
+        const usernameExists = await checkUsername();
+        if (!email || !username || !newUsername) {
+            setMessage('fill in all required information')
+            return;
+        }
+        if (usernameExists) {
+            setMessage("username already exists")
+            return;
+        }
+        try {
+            const res = await axios.put(`http://localhost:8080/users/${username}`, {
```

Comment:
the url you are passing in does not relate to any api endpoint you have defined in the username routes, you'll have to change the url to match your endpoint

Reply:
<img width="732" alt="image" src="https://github.com/user-attachments/assets/37d50fdd-0613-4791-ab87-c9972091d2f8">
<img width="360" alt="image" src="https://github.com/user-attachments/assets/80dafbcd-1811-41cd-9c0b-cecb8ab79046">
is this good?

Reply:
and this whoops
<img width="445" alt="image" src="https://github.com/user-attachments/assets/e7e03acd-ba0f-4f85-9e36-d19c1e442b89">